discovery/http.go

```diff
@@ -49,13 +52,24 @@ func init() {
 			return net.DialTimeout(n, a, defaultDialTimeout)
 		},
 	}
+	tInsecureTls := &http.Transport{
```

Comment:
You could probably do a shallow copy of `http.DefaultTransport` and modify the `TLSClientConfig` field.

```go
tInsecureTls := *http.DefaultTransport
tInsecureTls.TLSClientConfig = &tls.Config{InsecureSkipVerify: true}
ClientInsecureTls = &http.Client{
	Transport: &tInsecureTls,
}
```